### PR TITLE
Fix JQuery, Bluebird, and RSVP for TS3.9

### PR DESCRIPTION
**Pull request by rbuckton**
In TypeScript 3.9 we will be introducing a new awaited type operator along with updates to the definition of `Promise` and `PromiseLike` that cause incompatibilities with the definitions in JQuery, Bluebird, and RSVP. This PR adds a "typesVersions" redirection for TS 3.9+ for each package in advance of the upcoming TypeScript 3.9 beta.

NOTE: This PR merges the work from #43282, #43281, and #43280 as they each have build failures that are resolved by the other PRs.

Closes #43282
Closes #43281
Closes #43280

### CODE REVIEW

File: types/jquery/ts3.9/misc.deferred.d.ts [hunk 1]
```diff
@@ -0,0 +1,1125 @@
...
+            RRD = never, RJD = never, RND = never,
+            ARF = never, AJF = never, ANF = never,
+            BRF = never, BJF = never, BNF = never,
+            CRF = never, CJF = never, CNF = never,
+            RRF = never, RJF = never, RNF = never,
+            ARP = never, AJP = never, ANP = never,
+            BRP = never, BJP = never, BNP = never,
+            CRP = never, CJP = never, CNP = never,
+            RRP = never, RJP = never, RNP = never>(
+                doneFilter: (t: awaited TR, u: UR, v: VR, ...s: SR[]) => PromiseBase<ARD, AJD, AND,
```
Comment (Jessidhia) [inline types/jquery/ts3.9/misc.deferred.d.ts]:
```
../jquery/ts3.9/misc.deferred.d.ts(721,30): error TS2502: 't' is referenced directly or indirectly in its own type annotation.
```

Looks like a bug? The same appears in the other uses of `t: awaited TR`.

Comment (rbuckton) [reply types/jquery/ts3.9/misc.deferred.d.ts]: I'm looking into what's causing this.

**Comment by rbuckton** [comment]
I plan to revert the changes to JQuery with respect to `PromiseBase`. The issue with JQuery is *not* that it doesn't support `awaited`, but that someone else changed `PromiseBase` to be assignable to `Promise` so that people could substitute it. The problem is that JQuery's `PromiseBase` (and in reality, its `Deferred` implementation) is *not* Promise/A+ compatible. The [error](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43337#pullrequestreview-380168712) @Jessidhia mentioned above comes from the fact that JQuery's `Animation` is a "recursive promise":

```ts
    interface Animation<TElement> extends Promise3<
        Animation<TElement>, Animation<TElement>, Animation<TElement>,
        true | undefined, false, number,
        never, never, number
    >
```

The first type argument to `Promise3` is passed on to `PromiseBase` as the `TR` argument (which would have been recursively unwrapped in a Promise/A+ compatible promise). However, JQuery's `Deferred` doesn't unwrap. By trying to make `TR` an awaited type in `PromiseBase`, this results in an invalid recursive promise definition.

The actual change I need to make is to *break* the current expectation that `PromiseBase` is assignable to `Promise`, because they behave very differently in reality.

**Comment by rbuckton** [comment]
This last change may require #37570.

**Comment by rbuckton** [comment]
Closing this for now while we continue to investigate the impact of `awaited`. Non-`awaited`-related fixes will be moved to their own PR.

**Pull request closed without merging**